cachito/workers/pkg_managers/pip.py

```diff
@@ -1921,3 +1926,138 @@ def get_index_url(nexus_pypi_hosted_repo_url, username, password):
+    """
+    Resolve and fetch pip dependencies for the given app source archive.
+
+    :param Path path: the full path to the application source code
+    :param dict request: the Cachito request to resolve pip dependencies for
+    :param list requirement_files: a list of str representing paths to the Python requirement files
+        to be used to compile a list of dependencies to be fetched
+    :param list build_requirement_files: a list of str representing paths to the Python build
+        requirement files to be used to compile a list of build dependencies to be fetched
+    :return: a dictionary that has the following keys:
+        ``package`` which is the dict representing the main Package,
+        ``dependencies`` which is a list of dicts representing the package Dependencies
+    :rtype: dict
+    :raises CachitoError: if the package is not cachito-pip compatible
+    """
+    log.debug("Checking if the application source uses pip")
+    try:
+        pkg_name, pkg_version = get_pip_metadata(path)
+    except CachitoError:
+        log.exception("The requested package is not pip compatible")
+        raise
+
+    # This could be an empty list
+    if requirement_files is None:
+        requirement_files = _default_requirement_file_list(path)
+    else:
+        requirement_files = _get_absolute_pkg_file_paths(path, requirement_files)
+
+    # This could be an empty list
+    if build_requirement_files is None:
+        build_requirement_files = _default_requirement_file_list(path, devel=True)
+    else:
+        build_requirement_files = _get_absolute_pkg_file_paths(path, build_requirement_files)
+
+    requires = _download_from_requirement_files(request["id"], requirement_files)
+    buildrequires = _download_from_requirement_files(request["id"], build_requirement_files)
+
+    # Mark all build dependencies as Cachito dev dependencies
+    for dependency in buildrequires:
+        dependency["dev"] = True
+
+    # Upload dependencies and prepare return value
+    dependencies = []
+    pip_repo_name = get_pypi_hosted_repo_name(request["id"])
+    raw_repo_name = get_raw_hosted_repo_name(request["id"])
+    for requirement in requires + buildrequires:
+        dep = _push_downloaded_requirement(requirement, pip_repo_name, raw_repo_name)
+        dependencies.append(dep)
+
+    return {
+        "package": {"name": pkg_name, "version": pkg_version, "type": "pip"},
+        "dependencies": dependencies,
+    }
+
+
+def _get_absolute_pkg_file_paths(path, relative_paths):
+    """
+    Get the absolute paths for a list of files in a package.
+
+    :param Path path: the root path for the package containing the files in relative_paths
+    :param list relative_paths: paths to be converted to absolute paths
+    :return: absolute paths for the items in relative_paths
+    :rtype: list
+    """
+    # account for empty lists
+    if relative_paths is not None:
```

Comment:
I don't think we need to check for None here, this function is only called if the paths are not None